src/formatDate.js

```diff
@@ -7,8 +7,61 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newSeparator = toFormat[toFormat.length - 1];
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+
+  const dateParts = date.split(oldSeparator);
+
+  const yearIndexFrom =
+    fromFormat.indexOf('YYYY') !== -1
+      ? fromFormat.indexOf('YYYY')
+      : fromFormat.indexOf('YY');
+  const yearIndexTo =
+    toFormat.indexOf('YYYY') !== -1
+      ? toFormat.indexOf('YYYY')
+      : toFormat.indexOf('YY');
+  const monthIndexFrom = fromFormat.indexOf('MM');
+  const dayIndexFrom = fromFormat.indexOf('DD');
+
+  let year = dateParts[yearIndexFrom];
+  const month = dateParts[monthIndexFrom];
+  const day = dateParts[dayIndexFrom];
+
+  if (fromFormat[yearIndexFrom] === 'YY' && toFormat[yearIndexTo] === 'YYYY') {
+    if (year < 30) {
+      year = `20${year}`;
+    } else {
+      year = `19${year}`;
+    }
+  }
+
+  if (fromFormat[yearIndexFrom] === 'YYYY' && toFormat[yearIndexTo] === 'YY') {
+    year = year.slice(-2);
+  }
+
+  let newDate = '';
+
+  for (const part of toFormat.slice(0, -1)) {
```

Comment:
Using a `for...of` loop to iterate over the `toFormat` array and build the `newDate` string is not the most efficient approach. Instead, you could create an array to hold the date parts in the new order and then join them using the new separator. This would make the code cleaner and more efficient.